pkg/virt-handler/vm.go

```diff
@@ -499,6 +508,30 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 		condManager.RemoveCondition(vmi, v1.VirtualMachineInstanceAgentConnected)
 	}
 
+	// sync vmi guest time if aganet is availble and remove the required time sync condition
+	if condManager.HasCondition(vmi, v1.VirtualMachineInstanceRequireTimeSync) {
+		timeSyncRequestTimeoutInSeconds := 5.0
+		isConditionRemoved := false
+		if channelConnected {
+			err := d.setVMIGuestTime(vmi)
+			if err == nil {
+				condManager.RemoveCondition(vmi, v1.VirtualMachineInstanceRequireTimeSync)
+			}
+		}
+		if !isConditionRemoved {
+			var timeDiff time.Duration
+			for _, c := range vmi.Status.Conditions {
+				if c.Type == v1.VirtualMachineInstanceRequireTimeSync {
+					timeDiff = v12.Now().Sub((c.LastProbeTime).Time)
+					break
+				}
+			}
+			if timeDiff.Seconds() > timeSyncRequestTimeoutInSeconds {
```

Comment:
not sure if I understand what's happening here... if we don't manage to sync the time within 5 seconds after migration completed, we give up?

Reply:
:-) yes. 
I mean, I'm not sure about the time here. But I think we should give up at some point. I'm thinking that the agent should be available right after the migration is completed, there is no reason for us not to succeed. So if we didn't succeed, we probably won't succeed later too. I may be wrong, but I couldn't come up with anything better.